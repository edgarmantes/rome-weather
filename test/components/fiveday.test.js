import React from 'react';
import { Provider } from 'react-redux';
import FiveDayContainer, { FiveDay } from '../../src/components/fiveday';
import renderer from 'react-test-renderer';

import configureStore from 'redux-mock-store';

describe('<FiveDayContainer />' , () => {
	const initialState ={
		fiveDay : { daysList: [] },
		unit : "C",
		changeUnit : { unit : { props : { children : "F"}}}
	}
	const mockStore = configureStore();
	let store,wrapper

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = shallow( <FiveDayContainer store={store} /> )
    })

    it('Render the connected(SMART) component', () => {
       expect(wrapper.length).toEqual(1)
    });

	it('Renders FiveDay component properly', () => {
		shallow(<FiveDay dispatch={jest.fn()} />)
	});

	it("Renders with correct state.fiveDay.daysList value", () => {
		expect(wrapper.node.props.store.getState().fiveDay.daysList).toBe(initialState.fiveDay.daysList)
	})

	it("Renders with correct state.unit value", () => {
		expect(wrapper.node.props.store.getState().unit).toBe(initialState.unit)
	})

	it("Renders with correct state.getAlerts.alerts value", () => {
		expect(wrapper.node.props.store.getState().changeUnit.unit.props.children).toBe(initialState.changeUnit.unit.props.children)
	})

	it('Snapshot of FiveDay', () => {
        const renderedValue =  renderer.create(<FiveDay />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});
