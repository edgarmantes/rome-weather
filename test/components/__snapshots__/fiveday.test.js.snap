// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiveDayContainer /> Snapshot of FiveDay 1`] = `
<div
  className="fiveday-container col-6"
>
  <ul>
    Waiting...
  </ul>
</div>
`;
